core-java-modules/core-java-io-apis-2/src/main/java/com/baeldung/scanner/ScannerNoSuchElementException.java

```diff
@@ -0,0 +1,21 @@
+package com.baeldung.scanner;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class ScannerNoSuchElementException {
+
+    public static String readFileV1(String file) throws FileNotFoundException {
+        final Scanner scanner = new Scanner(new File(file));
+
+        return scanner.nextLine();
+    }
+
+    public static String readFileV2(String file) throws FileNotFoundException {
+        final Scanner scanner = new Scanner(new File(file));
+
+        return scanner.hasNextLine() ? scanner.nextLine() : "";
+    }
```

Comment:
We can improve this code using try-with-resources.

Reply:
done, I replaced also the legacy class File with Path